### PR TITLE
Like video Functionality

### PR DESCRIPTION
**Pull request by vijaytembugade**
- Like video
- remove from like video
- liked video list
- likeVideo context

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -49,6 +50,14 @@ function App() {
                 </PrivateRoutes>
               }
             />
+            <Route
+              path="/likes"
+              element={
+                <PrivateRoutes>
```
Comment (horsemaker) [inline src/App.js:56]: If you can name it `RequireAuth`, it will be according to the convention.

File: src/Components/VideoCard/VideoCard.js [hunk 2]
```diff
@@ -24,21 +30,84 @@ const VideoCard = ({ video }) => {
         icon: "😊",
       });
   }
+
+  async function handleLike(e) {
+    e.stopPropagation();
```
Comment (horsemaker) [inline src/Components/VideoCard/VideoCard.js:35]: Good use of `e.stopPropagation()`. 👍

File: src/Components/VideoCard/VideoCard.js [hunk 3]
```diff
@@ -24,21 +30,84 @@ const VideoCard = ({ video }) => {
...
+      toast.error(error.message);
+    }
+  }
+
+  async function handleDisLike(e) {
+    e.stopPropagation();
+    try {
+      if (isLoggedIn) {
+        const response = await removeFromLike(video._id, token);
+        console.log(response);
```
Comment (horsemaker) [inline src/Components/VideoCard/VideoCard.js:61]: remove `console.log(response)`

File: src/Contexts/likeContext/likeContext.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      (async () => {
+        try {
+          const response = await getLikedVideos(token);
+          setLikedVideos(response.data.likes);
+        } catch (error) {
+          toast.error("Failed to fetch liked videos");
+        }
+      })();
+    } else {
+      setLikedVideos([]);
```
Comment (horsemaker) [inline src/Contexts/likeContext/likeContext.js:25]: Nice check here!

File: src/index.js [hunk 5]
```diff
@@ -15,7 +20,9 @@ ReactDOM.render(
       <AuthProvider>
         <VideoProvider>
           <PlayListProvider>
-            <App />
+            <LikedVideoProvider>
+              <App />
+            </LikedVideoProvider>
           </PlayListProvider>
         </VideoProvider>
       </AuthProvider>
```
Comment (horsemaker) [inline src/index.js:28]: You could make a different file named `CombinedProviders` & put all your Providers there instead. The `index.js` file would look clean.

**Review by horsemaker — APPROVED** [review]

**Review by preethasri — APPROVED** [review]
good to merge

**Pull request merged**